source/MaterialXGenShader/Util.cpp

```diff
@@ -588,6 +588,47 @@ bool hasElementAttributes(OutputPtr output, const StringVec& attributes)
     return false;
 }
 
+bool inputChangeRequiresShaderGen(InputPtr input, const string& target)
```

Comment:
I think it would be worthwhile to clarify this function name a bit further, and one suggestion that comes to mind is `inputValueAffectsShaderGen`. This seems slightly closer to the core meaning of the function within MaterialXGenShader, and might make it more understandable and maintainable for new readers.  What are your thoughts?